integration_test.go

```diff
@@ -86,19 +124,222 @@ func testCartographerInternalState(t *testing.T, svc slam.Service, dataDir strin
+	if err := os.Mkdir(internalStateDir, 0o755); err != nil {
+		t.Error("failed to create test internal state directory")
+	}
+	filename := filepath.Join(internalStateDir, "map_data_"+timeStamp.UTC().Format(dataprocess.SlamTimeFormat)+".pbstream")
+	if err := os.WriteFile(filename, internalState, 0o644); err != nil {
+		t.Error("failed to write test internal state")
+	}
+}
+
+func testHelperCartographerFullMod(
+	t *testing.T,
+	dataDirectory string,
+	subAlgo viamcartographer.SubAlgo,
+	logger golog.Logger,
+	replaySensor bool,
+	mapRateSec int,
+	expectedMode cartofacade.SlamMode,
+) []byte {
+	t.Helper()
+	termFunc := internaltesthelper.InitTestCL(t, logger)
+	defer termFunc()
+
+	attrCfg := &vcConfig.Config{
+		Sensors: []string{"stub_lidar"},
+		ConfigParams: map[string]string{
+			"mode": reflect.ValueOf(subAlgo).String(),
+		},
+		MapRateSec:              &mapRateSec,
+		DataDirectory:           dataDirectory,
+		ModularizationV2Enabled: &_true,
+	}
+
+	done := make(chan struct{})
+	sensorReadingInterval := time.Millisecond * 200
+	timedSensor, err := testhelper.FullModIntegrationLidarTimedSensor(t, attrCfg.Sensors[0], replaySensor, sensorReadingInterval, done)
+	test.That(t, err, test.ShouldBeNil)
+
+	svc, err := internaltesthelper.CreateFullModSLAMServiceIntegration(t, attrCfg, timedSensor, logger)
+	test.That(t, err, test.ShouldBeNil)
+	start := time.Now()
+
+	cSvc, ok := svc.(*viamcartographer.CartographerService)
+	test.That(t, ok, test.ShouldBeTrue)
+	test.That(t, cSvc.SlamMode, test.ShouldEqual, expectedMode)
+
+	ctx, cancelFunc := context.WithTimeout(context.Background(), time.Second*5)
+
+	defer cancelFunc()
+
+	// wait till all lidar readings have been read
+	if !utils.SelectContextOrWaitChan(ctx, done) {
+		test.That(t, errors.New("test timeout"), test.ShouldBeNil)
+	}
+
+	testCartographerFullModPosition(t, svc, attrCfg.Sensors[0])
+	testCartographerMap(t, svc, cSvc.SlamMode == cartofacade.LocalizingMode)
+
+	internalState, err := slam.GetInternalStateFull(context.Background(), svc)
+	test.That(t, err, test.ShouldBeNil)
+
+	// Close out slam service
+	test.That(t, svc.Close(context.Background()), test.ShouldBeNil)
+
+	stop := time.Now()
+	testDuration := stop.Sub(start)
+	t.Logf("test duration %d", testDuration)
+
+	// Check that a map was generated as the test has been running for more than the map rate msec
+	mapsInDir, err := os.ReadDir(path.Join(dataDirectory, "internal_state"))
+	test.That(t, err, test.ShouldBeNil)
+	test.That(t, testDuration.Seconds(), test.ShouldBeGreaterThanOrEqualTo, time.Duration(*attrCfg.MapRateSec).Seconds())
+	test.That(t, len(mapsInDir), test.ShouldBeGreaterThan, 0)
+
+	// return the internal state so updating mode can be tested
+	return internalState
+}
+
+func integrationtestHelperCartographerFullMod(t *testing.T, subAlgo viamcartographer.SubAlgo) {
+	t.Helper()
+	logger := golog.NewTestLogger(t)
+	t.Run("live sensor mapping mode", func(t *testing.T) {
```

Comment:
@JohnN193 & @EshaMaharishi do these integration tests make sense to y'all for the cloud story? I've removed tests that refer to live mode (as live mode won't exist post full mod), 

They test the cartesian product of 
slam_mode = [mapping,updating,localizing]
sensor_type = [live, replay]

Reply:
I think so, and just to confirm
1. you have a bool to signal whether to use a replay camera
2. for updating mode and localizing mode, you do an initial run to generate an internal state, then you run the test in the desired mode

Only question I have is why not use an artifact for the internal state?

Reply:
1. correct
2. correct 

Because the pre full mod tests tests used the internal state from the previous offline run. This is keeping the same behavior (and also proving that the internal state we get from the GetInternalState api is able to be used to initialize cartographer.